### PR TITLE
feat(refactor): parse_nist_parameters

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### This PR fixes:

- Refactor parsing of nist parameters

### How to test:

- Automatic and continuous test pass

### Expected outcome:
- Installation, unit and integration tests pass

### Review:
- [ ] Code review
- [ ] New code is executed and covered by tests
- [ ] Tests pass

### CODE REVIEW

**Comment by henrikstranneheim** [comment]
@jemten It got i bit big, but I have not really changed much - just moved it around

File: lib/MIP/Reference.pm [hunk 1]
```diff
@@ -177,6 +183,321 @@ sub check_human_genome_file_endings {
...
+
+        # Alias
+        my $nist_href = \%{ $active_parameter_href->{$nist_parameter} };
+
+      NIST_VERSION:
+        foreach my $nist_version ( keys %{$nist_href} ) {
+
+          NIST_FILE:
+            while ( my ( $nist_id, $file_path ) = each %{ $nist_href->{$nist_version} } )
+            {
```
Comment (jemten) [inline lib/MIP/Reference.pm:234]: It looks like this part of the sub is repeated in several of the subs (`set_nist_file_name_path`, `check_nist_file_name` aside from this one). Is it possible to write a separate sub for that kind of parsing, or do both file_name_check, file_path_check and file_path_set in one loop?

File: lib/MIP/Reference.pm [hunk 2]
```diff
@@ -177,6 +183,321 @@ sub check_human_genome_file_endings {
...
+
+            }
+        }
+    }
+    return 1;
+}
+
+sub check_nist_nist_id {
+
+## Function : Check nist_ids hash contains supplied nist_id in nist parameters
```
Comment (jemten) [inline lib/MIP/Reference.pm:316]: that's a lot of nist 😄

Comment (henrikstranneheim) [reply lib/MIP/Reference.pm:250]: yeah

File: lib/MIP/Reference.pm (hunk 1 above)
Comment (henrikstranneheim) [reply lib/MIP/Reference.pm]: Have made another attempt in https://github.com/Clinical-Genomics/MIP/pull/1456/commits/916273887a9ba6edf35e0b6f3ab6418bddac8dfd. Please have a look

**Pull request merged**